### PR TITLE
M2 doesn't use very short stubs of clipped reads for genotyping

### LINKED ISSUE #5060: M2 getting thrown off by clipped read stubs
Suppose the reference haplotype is
TAAGC. . . . TAAGG. . . .

and an alt haplotype (SNV at the last G shown) is
TAAGC. . . TAAG**C**. . .

Suppose further that we have a read ending in the first TAAGC that has been hard-clipped (to fit the assembly region) to just a 5-base TAAGC stub.

Pair-HMM is fully Bayesian and computes the total likelihood of *all* possible alignments of a read to each haplotype.  This gives the alt haplotype a factor of 2 advantage because TAAGC matches it in two locations, so there are two perfectly matching alignments instead of one.  In log 10 space this is log_10(2) = 0.301, which is greater than our 0.2 threshold for a likelihood to be considered informative.

Therefore, by clipping the read and losing the information of its first 96 bases, we end up considering it informative for the wrong haplotype.  This can lead to false positives.  It *also* causes false negatives because sometimes a read stub from the normal sample get misaligned to the alt haplotype, triggering the normal artifact filter.

It also causes problems in our bamout-based MC3 validation.

### PR DESCRIPTION
**Pull request by davidbenjamin**
Closes #5060.

@meganshand This fixes your bug.  Do you have time to review before Monday's release?

@takutosato @LeeTL1220 It improves sensitivity and specificity.

@ldgauthier This probably affects HaplotypeCaller as well.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java [hunk 1]
```diff
@@ -60,6 +60,9 @@
     public static final int MAX_NORMAL_QUAL_SUM = 100;
     public static final int MIN_PALINDROME_SIZE = 5;
 
+    // after trimming to fit the assembly window, throw away read stubs shorter than this length
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java]: Could you add why you do this and cite the issue in github?

**Comment by meganshand** [comment]
@davidbenjamin This filter looks fine to me, but could you add a test? You should have the data from the mitochondria bug already, right? Also if this potentially affects HaplotypeCaller should this fix be somewhere deeper so the filter is used by both tools or would we want the fix to be different?

**Comment by davidbenjamin** [comment]
Back to @meganshand. I put in a simple mitochondrial integration test.  Given that our MC3 validation already covers this particular bug I actually don't think it needs a new test for mitochondria.  Also, for later, are any of your spike-in bams public (or rather, public + public)?  I noticed that the NA12878 truth doesn't have very low AFs.

**Comment by meganshand** [comment]
@davidbenjamin Thanks for the test! Unfortunately none of the spike-in bams I have are public, but I will ask Sarah Calvo if she knows of any samples that would work as a spike-in and are public. Maybe we can track one down.

**Pull request merged**